Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/MergeWorkspacesWithLimits.py

```diff
@@ -0,0 +1,119 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+from __future__ import (absolute_import, division, print_function)
+
+from mantid.simpleapi import (ConjoinWorkspaces, CropWorkspaceRagged, DeleteWorkspace, MatchSpectra, Rebin, SumSpectra)
+from mantid.api import (DataProcessorAlgorithm, AlgorithmFactory, WorkspaceProperty, )
+from mantid.kernel import (Direction, FloatArrayProperty)
+import numpy as np
+
+
+class MergeWorkspacesWithLimits(DataProcessorAlgorithm):
+
+    def __init__(self):
+        """Initialize an instance of the algorithm."""
+        DataProcessorAlgorithm.__init__(self)
+
+    def name(self):
+        return 'MergeWorkspacesWithLimits'
+
+    def category(self):
+        return 'Workflow\\Diffraction'
+
+    def seeAlso(self):
+        return []
+
+    def summary(self):
+        return 'Merges a group workspace using weighting from a set of range limits for each workspace.'
+
+    def checkGroups(self):
+        return False
+
+    def version(self):
+        return 1
+
+    def validateInputs(self):
+        # given workspaces must exist
+        # and must be public of ExperimentInfo
+        issues = dict()
+        ws_group = self.getProperty('WorkspaceGroup').value
+        x_min = self.getProperty('XMin').value
+        x_max = self.getProperty('XMax').value
+        if not x_min.size == ws_group.size():
+            issues['x_min'] = 'x_min entries does not match size of workspace group'
+        if not x_max.size == ws_group.size():
+            issues['x_max'] = 'x_max entries does not match size of workspace group'
+        return issues
+
+    def PyInit(self):
+        self.declareProperty(WorkspaceProperty('WorkspaceGroup', '', direction=Direction.Input),
```

Comment:
If we are going to be trying to make it more useful outside of Polaris it may also be worth me seeing if it can be made to work with workspaces with multiple spectra as well as group workspaces while I am at it.

Reply:
That would also make a difference what the algorithm is eventualy named